### PR TITLE
Testing listing model - all test passing

### LINKED ISSUE #6: Testing for the models - User story 1

### PR DESCRIPTION
**Pull request by Drescoding**
closes #6 

Validating the Listings model.

### CODE REVIEW

**Review by fbl11 — CHANGES_REQUESTED** [review]
line 9: change to listing

File: test/models/listing_test.rb [hunk 1]
```diff
@@ -1,7 +1,43 @@
...
 class ListingTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  def setup
+    @listing = Listing.new(property_name: "Big house", description: "Nice bedrooms", price: 100, start_date:"2018-10-15")
+  end
+
+
+  test "should be an instance of a user" do
```
Comment (fbl11) [inline test/models/listing_test.rb]: change user to listing

File: test/models/listing_test.rb [hunk 2]
```diff
@@ -1,7 +1,43 @@
...
+    @listing = Listing.new(property_name: "Big house", description: "Nice bedrooms", price: 100, start_date:"2018-10-15")
+  end
+
+
+  test "should be an instance of a listing" do
+    assert @listing.valid?
+  end
+
+  test "should not create a new listing without a name" do
+    listing2 = Listing.create(description: "Nice bedrooms", price: 100, start_date:"2018-10-15")
```
Comment (fbl11) [inline test/models/listing_test.rb:14]: refactor to @listing

**Review by fbl11 — APPROVED** [review]
line 9: change to listing

File: test/models/listing_test.rb [hunk 3]
```diff
@@ -1,7 +1,43 @@
...
+  test "should not create a new listing without a date" do
+    @listing.start_date = ''
+    assert_not @listing.valid?
+  end
+
+  test "the price should be a number" do
+    @listing.price = "notanumber"
+    assert_not @listing.valid?
+  end
+
```
Comment (AlexDresco) [inline test/models/listing_test.rb]: Well done!

**Review by Drescoding — COMMENTED** [review]
Tada!

File: test/models/listing_test.rb [hunk 4]
```diff
@@ -1,7 +1,43 @@
 require 'test_helper'
 
 class ListingTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  def setup
+    @listing = Listing.new(property_name: "Big house", description: "Nice bedrooms", price: 100, start_date:"2018-10-15")
+  end
+
```
Comment (Drescoding) [inline test/models/listing_test.rb:7]: extra blank lines

File: app/models/listing.rb [hunk 5]
```diff
@@ -4,5 +4,5 @@ class Listing < ApplicationRecord
   validates(:description, presence: true)
   validates(:price, presence: true)
   validates(:start_date, presence: true)
-
```
Comment (AlexDresco) [inline app/models/listing.rb]: Hello

File: test/models/listing_test.rb [hunk 6]
```diff
@@ -1,7 +1,43 @@
...
+  test "should not create a new listing without a description" do
+    listing2 = Listing.create(property_name: "Big house", price: 100, start_date:"2018-10-15")
+    assert_not listing2.valid?
+  end
+
+  test "should not create a new listing without a price" do
+    listing2 = Listing.create(property_name: "Big house", description: "Nice bedrooms", start_date:"2018-10-15")
+    assert_not listing2.valid?
+  end
+
```
Comment (AlexDresco) [inline test/models/listing_test.rb]: Well done

**Pull request merged**